integrations/Braze/browser.js

```diff
@@ -172,28 +147,121 @@ class Braze {
       "gender",
       "lastname",
       "phone",
-      "facebook",
-      "twitter",
-      "first_name",
-      "last_name",
       "dob",
       "external_id",
       "country",
       "home_city",
       "bio",
-      "gender",
-      "phone",
       "email_subscribe",
       "push_subscribe",
     ];
 
+    // deep clone the traits object
+    const traits = cloneDeep(rudderElement.message.context.traits);
+
     reserved.forEach((element) => {
       delete traits[element];
     });
+    if (this.supportDedup && this.previousPayload !== null) {
+      const prevUserId = get(this.previousPayload, "rudderElement.message.userId");
+      const prevAddress = get(
+        this.previousPayload,
+        "rudderElement.message.context.traits.address"
+      );
+      const prevBirthday = get(
+        this.previousPayload,
+        "rudderElement.message.context.traits.birthday"
+      );
+      const prevEmail = get(
+        this.previousPayload,
+        "rudderElement.message.context.traits.email"
+      );
+      const prevFirstname = get(
+        this.previousPayload,
+        "rudderElement.message.context.traits.firstname"
+      );
+      const prevGender = get(
+        this.previousPayload,
+        "rudderElement.message.context.traits.gender"
+      );
+      const prevLastname = get(
+        this.previousPayload,
+        "rudderElement.message.context.traits.lastname"
+      );
+      const prevPhone = get(
+        this.previousPayload,
+        "rudderElement.message.context.traits.phone"
+      );
 
-    Object.keys(traits).forEach((key) => {
-      window.braze.getUser().setCustomUserAttribute(key, traits[key]);
-    });
+      if (userId && userId !== prevUserId) {
+        window.braze.changeUser(userId);
+      }
+      if (address && address !== prevAddress) {
+        window.braze.getUser().setCountry(address.country);
+        window.braze.getUser().setHomeCity(address.city);
+      }
+      if (birthday && birthday !== prevBirthday) {
+        window.braze
+          .getUser()
+          .setDateOfBirth(
+            birthday.getUTCFullYear(),
+            birthday.getUTCMonth() + 1,
+            birthday.getUTCDate()
+          );
+      }
+      if (email && email !== prevEmail) {
+        window.braze.getUser().setEmail(email);
+      }
+      if (firstname && firstname !== prevFirstname) {
+        window.braze.getUser().setFirstName(firstname);
+      }
+      if (gender && gender !== prevGender) {
+        window.braze.getUser().setGender(this.formatGender(gender));
+      }
+      if (lastname && lastname !== prevLastname) {
+        window.braze.getUser().setLastName(lastname);
+      }
+      if (phone && phone !== prevPhone) {
+        window.braze.getUser().setPhoneNumber(phone);
+      }
```

Comment:
This part is repeating, can't we store them these attributes and use?

Reply:
yes, updated